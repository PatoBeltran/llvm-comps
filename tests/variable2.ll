; ModuleID = 'variable2.c'
target datalayout = "e-m:o-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-apple-macosx10.11.0"

; Function Attrs: nounwind ssp uwtable
define i64 @inter() #0 {
  %a = alloca i64, align 8
  %b = alloca i64, align 8
  %c = alloca i64, align 8
  %d = alloca i64, align 8
  store i64 10, i64* %a, align 8
  store i64 5, i64* %b, align 8
  store i64 4, i64* %c, align 8
  store i64 4, i64* %a, align 8
  %1 = load i64, i64* %a, align 8
  %2 = load i64, i64* %b, align 8
  %3 = add nsw i64 %1, %2
  store i64 %3, i64* %d, align 8
  store i64 5, i64* %a, align 8
  %4 = load i64, i64* %d, align 8
  %5 = load i64, i64* %c, align 8
  %6 = add nsw i64 %4, %5
  %7 = load i64, i64* %a, align 8
  %8 = add nsw i64 %6, %7
  %9 = load i64, i64* %b, align 8
  %10 = add nsw i64 %8, %9
  ret i64 %10
}

attributes #0 = { nounwind ssp uwtable "disable-tail-calls"="false" "less-precise-fpmad"="false" "no-frame-pointer-elim"="true" "no-frame-pointer-elim-non-leaf" "no-infs-fp-math"="false" "no-nans-fp-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="core2" "target-features"="+cx16,+fxsr,+mmx,+sse,+sse2,+sse3,+ssse3" "unsafe-fp-math"="false" "use-soft-float"="false" }

!llvm.module.flags = !{!0}
!llvm.ident = !{!1}

!0 = !{i32 1, !"PIC Level", i32 2}
!1 = !{!"Apple LLVM version 7.3.0 (clang-703.0.29)"}
