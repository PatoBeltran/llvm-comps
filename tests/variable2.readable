_inter:
  mov %rbx, 10
  mov %rcx, 5
  mov %rdx, 4
  mov %rbx, 4
  mov %rsi, %rcx
  add %rsi, %rbx
  mov %rdi, %rsi
  mov %rbx, 5
  mov %r8, %rdx
  add %r8, %rdi
  mov %r9, %rbx
  add %r9, %r8
  mov %r10, %rcx
  add %r10, %r9
  mov %rax, %r10
  ret

