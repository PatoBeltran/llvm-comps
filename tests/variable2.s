.intel_syntax noprefix
.text
    .globl _inter
    .type _inter, @function
  _inter:
    .cfi_startproc
    push rbp
    mov rbp, rsp
    mov rbx, 10
    mov r12, 5
    mov r13, 4
    mov rbx, 4
    mov r14, rbx
    add r14, r12
    mov r15, r14
    mov rbx, 5
    mov r13, r15
    add r13, r13
    mov rbx, r13
    add rbx, rbx
    mov r12, rbx
    add r12, r12
    mov rax, r12
    mov rsp, rbp
    pop rbp
    ret
    .cfi_endproc

  _inter.end:
    .size _inter, .-_inter

