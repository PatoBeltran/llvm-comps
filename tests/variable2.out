	.section	__TEXT,__text,regular,pure_instructions
	.macosx_version_min 10, 11
	.globl	_inter
	.p2align	4, 0x90
_inter:                                 ## @inter
	.cfi_startproc
## BB#0:
	pushq	%rbp
Ltmp0:
	.cfi_def_cfa_offset 16
Ltmp1:
	.cfi_offset %rbp, -16
	movq	%rsp, %rbp
Ltmp2:
	.cfi_def_cfa_register %rbp
	movq	$10, -16(%rbp)
	movq	$5, -8(%rbp)
	movq	$4, -32(%rbp)
	movq	$4, -16(%rbp)
	movq	-8(%rbp), %rax
	addq	$4, %rax
	movq	%rax, -24(%rbp)
	movq	$5, -16(%rbp)
	movq	-24(%rbp), %rax
	addq	-32(%rbp), %rax
	movq	-8(%rbp), %rcx
	leaq	5(%rcx,%rax), %rax
	popq	%rbp
	retq
	.cfi_endproc


.subsections_via_symbols
